src/main/java/org/opensearch/neuralsearch/rest/RestNeuralStatsAction.java

```diff
@@ -167,7 +168,7 @@ private NeuralStatsInput createNeuralStatsInputFromRequestParams(RestRequest req
         }
 
         // Process requested stats
-        boolean anyStatAdded = processRequestedStats(stats.get(), neuralStatsInput);
+        boolean anyStatAdded = processRequestedStats(request, stats.get(), neuralStatsInput);
```

Comment:
I think you may simplify code a little bit by combining line 161 to 176  into `void processRequestedStats(RestRequest request, NeuralStatsInput neuralStatsInput)`. 

Basically, you can change `void processRequestedStats(RestRequest request, NeuralStatsInput neuralStatsInput)` to do below steps:
1. get `Optional<String[]> stats` from request
2. 
    - 2a: if stats is not present, add all stats to neuralStatsInput and return
    - 2b: if stats is present, continue
3. Validate stats and add stats to `neuralStatsInput` as same as current logic
    -3a: Throw exception if invalidStatNames is not empty
    -3b: Call if necessary addAllStats(neuralStatsInput);
6.

Reply:
Makes sense, I think we can skip the 3b `addAllStats` call as well. When we start validating, we will either be adding specific requested stats (meaning we do not want to add all) or we will be throwing an exception because there are invalid stats (meaning we do not want to add all).